components/editor/utils/CodeEditorFields.js

```diff
@@ -78,31 +78,39 @@ export const PlainEditor = ({ label, value, onChange, onPaste, mode }) => (
 )
 
 export const JSONEditor = ({ label, value, onChange }) => {
-  const [stateValue, setStateValue] = useState(undefined)
+  const [stateValue, setStateValue] = useState(null)
+  const [chartType, setChartType] = useState(null)
+
+  useEffect(() => {
+    if (!stateValue || chartType !== value.type) {
+      const stringified = JSON.stringify(value, null, 2)
+      setStateValue(stringified)
+      setChartType(value.type)
+    }
+  }, [value])
```

Comment:
a more generic way:
```
import { usePrevious } from '@project-r/styleguide'
```


```suggestion
  const [stateValue, setStateValue] = useState(null)
  const previousValue = usePrevious(value)

  useEffect(() => {
    if (previousValue !== value) {
      const stringified = JSON.stringify(value, null, 2)
      setStateValue(stringified)
    }
  }, [value, previousValue])
```

Object identity would probably suffice. Otherwise you could also do a deeper compare: `if (previousValue || value.type !== previousValue.type)`